ecoscope/analysis/ecograph.py

```diff
@@ -75,10 +75,10 @@ def __init__(self, trajectory, resolution=15, radius=2, cutoff=None, tortuosity_
         def compute(df):
             subject_name = df.name
             print(f"Computing EcoGraph for subject {subject_name}")
-            G = self._get_ecograph(self, df, subject_name, radius, cutoff, tortuosity_length)
+            G = self._get_ecograph(df, subject_name, radius, cutoff, tortuosity_length)
             self.graphs[subject_name] = G
 
-        self.trajectory.groupby("groupby_col").progress_apply(compute)
+        self.trajectory.groupby("groupby_col")[self.trajectory.columns].progress_apply(compute)
```

Comment:
Why do we need to explictly add `[self.trajectory.columns]`?

Reply:
`include_groups=True` has been [deprecated](https://pandas.pydata.org/pandas-docs/version/2.2/reference/api/pandas.core.groupby.DataFrameGroupBy.apply.html) on `groupby`. Explicitly adding the columns preserves the existing behavior and effectively silences the deprecation warning.



We'll need to revisit this, it doesn't seem very inline with the direction Pandas is going. I'm looking at reworking all the classes in `base.py` to be [accessors](https://pandas.pydata.org/docs/reference/api/pandas.api.extensions.register_dataframe_accessor.html) (#118) and will probably tackle this as part of that work

Reply:
Got it. Thank you!